### PR TITLE
Functions to secure sorting of wallet list on wallet selectors

### PR DESCRIPTION
**Pull request by amanalili**
Task: Wallet List scene order of wallets is not the same as the wallet selector list. (https://app.asana.com/0/361770107085503/1154597586670346)

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [x] Tested on iOS Tablet
- [x] Tested on small Android

### CODE REVIEW

File: src/components/modals/WalletListModal.js [hunk 1]
```diff
@@ -53,17 +54,18 @@ export class WalletListModal extends Component<Props, LocalState> {
...
-        walletItem: wallet,
-        supportedWalletType: null
-      }
-      records.push(record)
-      totalCurrenciesAndTokens++
-      if (wallet.enabledTokens.length) {
-        totalCurrenciesAndTokens = totalCurrenciesAndTokens + wallet.enabledTokens.length
+    for (i; i < this.props.activeWalletIds.length; i++) {
+      const wallet = this.props.wallets.find(wallet => wallet.id === this.props.activeWalletIds[i])
+      if (wallet) {
```
Comment (paullinator) [inline src/components/modals/WalletListModal.js]: Need to add back the logic that excludes FIO

**Comment by amanalili** [comment]
> Change the WalletListModal to be connected so that you do not have to pass in the active Wallet IDs from all of the locations that use the modal

Fixed

File: src/components/modals/WalletListModal.js (hunk 1 above)
Comment (amanalili) [reply src/components/modals/WalletListModal.js]: Fixed

File: src/components/modals/WalletListModal.js [hunk 2]
```diff
@@ -14,7 +14,12 @@ import type { GuiWallet } from '../../types/types.js'
 import { scale } from '../../util/scaling.js'
 import { type AirshipBridge, AirshipModal } from './modalParts.js'
 
-type Props = {
+export type connectorProps = {
```
Comment (paullinator) [inline src/components/modals/WalletListModal.js]: This should be called StateProps

File: src/components/modals/WalletListModal.js [hunk 3]
```diff
@@ -14,7 +14,12 @@ import type { GuiWallet } from '../../types/types.js'
...
+  state: State,
+  activeWalletIds: Array<string>
+}
+
+type componentProps = {
```
Comment (paullinator) [inline src/components/modals/WalletListModal.js]: This should be called OwnProps

File: src/connectors/components/WalletListConnector.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import type { connectorProps } from '../../components/modals/WalletListModal'
+import { WalletListModal } from '../../components/modals/WalletListModal'
+import { getActiveWalletIds } from '../../modules/UI/selectors.js'
+import type { State } from '../../types/reduxTypes.js'
+
+export const mapStateToProps = (state: State): connectorProps => {
+  const wallets = state.ui.wallets.byId
+  const activeWalletIds = getActiveWalletIds(state).filter(id => !(wallets[id] != null && wallets[id].type === 'wallet:fio'))
+  return {
+    state,
```
Comment (paullinator) [inline src/connectors/components/WalletListConnector.js]: It looks like you no longer need the state. Please remove

File: src/connectors/components/WalletListConnector.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+// @flow
```
Comment (paullinator) [inline src/connectors/components/WalletListConnector.js]: This file should be called WalletListModalConnector

File: src/connectors/components/WalletListConnector.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  const activeWalletIds = getActiveWalletIds(state).filter(id => !(wallets[id] != null && wallets[id].type === 'wallet:fio'))
+  return {
+    state,
+    activeWalletIds
+  }
+}
+
+const WalletListConnector = connect(mapStateToProps)(WalletListModal)
+
+export { WalletListConnector }
```
Comment (paullinator) [inline src/connectors/components/WalletListConnector.js]: Change the export to WalletListModalConnected

File: src/components/modals/WalletListModal.js (hunk 2 above)
Comment (amanalili) [reply src/components/modals/WalletListModal.js]: Fix

File: src/components/modals/WalletListModal.js (hunk 3 above)
Comment (amanalili) [reply src/components/modals/WalletListModal.js]: Fixed

File: src/connectors/components/WalletListConnector.js (hunk 4 above)
Comment (amanalili) [reply src/connectors/components/WalletListConnector.js]: State is needed by the WalletListModal itself

File: src/connectors/components/WalletListConnector.js (hunk 5 above)
Comment (amanalili) [reply src/connectors/components/WalletListConnector.js]: Fixed

File: src/connectors/components/WalletListConnector.js (hunk 6 above)
Comment (amanalili) [reply src/connectors/components/WalletListConnector.js]: Fixed

File: src/connectors/components/WalletListConnector.js (hunk 4 above)
Comment (paullinator) [reply src/connectors/components/WalletListConnector.js]: WalletListModal does not appear to use state. From what i see it passes state into CryptoExchangeWalletListRow which is used to call the selector calculateWalletFiatBalance.  Selectors should not be called from within components. Instead you should make CryptoExchangeWalletListRow a connected component and use its connector to call calculateWalletFiatBalance and pass in the result.

Comment (amanalili) [reply src/connectors/components/WalletListConnector.js]: Fixed, updated PR

**Pull request merged**